File: src/interfaces/jobSource.js
'use strict';

// Strategy interface — every job board implements source_name + search().
class JobSource {
  get sourceName() {
    throw new Error('JobSource subclasses must implement sourceName');
  }

  // eslint-disable-next-line no-unused-vars
  async search(role, location, pages = 2) {
    throw new Error('JobSource subclasses must implement search()');
  }
}

module.exports = { JobSource };

File: python/src/interfaces/job_source.py
from abc import ABC, abstractmethod
from ..models import Job


class JobSource(ABC):
    @property
    @abstractmethod
    def source_name(self) -> str:
        pass

    @abstractmethod
    async def search(self, role: str, location: str, pages: int = 2) -> list[Job]:
        pass
